scrapscript.py

```diff
@@ -1190,6 +1231,8 @@ def bencode(obj: object) -> bytes:
     assert not isinstance(obj, bool)
     if isinstance(obj, int):
         return b"i" + str(int(obj)).encode("ascii") + b"e"
+    if isinstance(obj, float):
+        return b"f" + str(float(obj)).encode("ascii") + b"e"
```

Comment:
This is not standard bencode; probably we should serialize them as strings (they are tagged at a higher level)

Reply:
Or we should dump them as bytestrings or something since they're all 8 bytes wide

Reply:
Although, hmm, there is apparently an extension: https://github.com/mgorny/gx86-test/blob/master/net-p2p/bittorrent/files/bittorrent-4.4.0-bencode-float.patch

Reply:
Oh cool. Looks like they replace the e for scientific notation with a capital E to avoid clashing with the bencode end sentinel e

Reply:
Do you like that? Could also just not support float serialization for now

Reply:
I am tentatively on team not-serialize-for-now but if you want to go ahead and do the full rigamarole for porting that impl and working on tests and stuff, go for it

Reply:
We're on the same team. It's a shame to not be able to use floats in the online repl, but we can have a nice error message that says floats aren't supported rather than getting some cryptic syntax error like we get now when trying to use floats